src/utils/generateCombinations.test.ts

```diff
@@ -0,0 +1,62 @@
+import { generateCombinations } from "./generateCombinations";
+
+describe("generateCombinations", () => {
+  it("should generate all combinations of 1 empty array", () => {
+    const arrays = [[]];
+    const result = generateCombinations(arrays);
+    expect(result).toEqual([""]);
```

Comment:
why is the result not an empty array `[]`?

Reply:
No reason, it would probably be more logical to be empty indeed. 
Updated accordingly.